src/common/utils.h

```diff
@@ -215,6 +215,13 @@ void CheckFormatImpl(const RunContext &rctx, const NDArray &input,
 }
 
 
+template<typename xpu>
```

Comment:
Let's make sure functions in `.h` are documented. Should add some description for `CastStorageDispatch` too...